R/build-reference-index.R

```diff
@@ -42,23 +42,32 @@ data_reference_index_rows <- function(section, pkg) {
 
 
   if (has_name(section, "contents")) {
-    in_section <- select_topics(section$contents, pkg$topics)
-    topics <- pkg$topics[in_section, ]
-
-    contents <- tibble::tibble(
-      path = topics$file_out,
-      aliases = purrr::map2(
-        topics$funs,
-        topics$name,
-        ~ if (length(.x) > 0) .x else .y
-      ),
-      title = topics$title,
-      icon = find_icons(topics$alias, path(pkg$src_path, "icons"))
-    )
-
+    any_not_string <- any(purrr::map_chr(section$contents, typeof) != "character")
+    any_no_length <- any(!nzchar(section$contents))
+    if (any_not_string || any_no_length) {
+      abort(
```

Comment:
Can you please rewrite this to eliminate the very deep nesting? (By pulling out subexpressions into named variables)

Reply:
I've tried simplifying a bit more.